applications/OptimizationApplication/python_scripts/controls/master_control.py

```diff
@@ -0,0 +1,173 @@
+            # iterate through each control to create its own container expression map from the collective expressions map given as input
+            control_physical_sensitivities_container_expression_map = {}
+            for physical_control_variable in control.GetPhysicalKratosVariables():
+                # get the required physical variables from control.
+                if physical_control_variable in physical_space_gradient_variable_and_collective_expressions_map.keys():
+                    # if the sensitivities for the given physical control variable exists, then try to find whether
+                    # for this specific control the physical control variable sensitivities exists.
+                    control_expression = control.GetEmptyControlField()
+                    sensitivity_collective_expression = physical_space_gradient_variable_and_collective_expressions_map[physical_control_variable]
+                    for container_expression in sensitivity_collective_expression.GetContainerExpressions():
+                        if control_expression.GetContainer() == container_expression.GetContainer():
+                            # there exists for this control's physical variables sensitivities. then add it to control map.
+                            control_expression.CopyFrom(container_expression)
+                            control_physical_sensitivities_container_expression_map[physical_control_variable] = control_expression
+                            break
+
+            # map the physical control variable sensitivities to one control space
+            mapped_gradients.Add(control.MapGradient(control_physical_sensitivities_container_expression_map))
+
+        return mapped_gradients
+
+    def Update(self, update_collective_expressions: KratosOA.ContainerExpression.CollectiveExpressions) -> 'dict[Control, bool]':
+        """Update each control with given collective expression's respective container expression.
+
+        Args:
+            update_collective_expressions (KratosOA.ContainerExpression.CollectiveExpressions): Update
+
+        Raises:
+            RuntimeError: If number of controls and number of container expressions mismatch.
+
+        Returns:
+            dict[Control, bool]: A map with control and a boolean whether the update changed anything in that control.
+        """
+        if len(self.__list_of_controls) != len(update_collective_expressions.GetContainerExpressions()):
+            raise RuntimeError(f"Controls size and update size mismatch [ number of controls: {len(self.__list_of_controls)}, number of container expressions: {len(update_collective_expressions.GetContainerExpressions())} ].")
+
+        update_map: 'dict[Control, bool]' = {}
+        for control, container_expression in zip(self.__list_of_controls, update_collective_expressions.GetContainerExpressions()):
+            update_map[control] = control.Update(container_expression)
+
+        return update_map
+
```

Comment:
I have found out that we need a new method, which is GetEmptyControlField(). The function should go through all controls and call GetEmptyControlField() for each one. The function should return collective expression.

For VM control GetEmptyControlField() should return zeros. For the rest ones, I don't know=), I guess in topology we take directly the Young's Modulus and densitiy?

Reply:
If you need the initial control field, what you can do is, call `GetEmptyControlField` of the `MasterControl` at the initialize, and save it somewhere you want. I would keep it inside the algorithm because, this requiremenet is algorithm specific. What would you say?

Reply:
That is the point, that I have tried to explain. You can use GetEmptyControlField to Init the control field if we use Vertex Morphing. In the case of topology optimization, I think we need to get the Young Module of the Model as an Initial control field.

This comment is not a blocker. For now, I can code with GetEmptyControlField. If for Topology we need smth else, we can add a function and adjust the algorithm's Initial.